omod/src/main/webapp/resources/scripts/fragments/dashboardwidgets/relationships/relationships.controller.js

```diff
@@ -97,25 +112,41 @@ function RelationshipsController(openmrsRest, $scope) {
             q: searchString,
             v: 'custom:(uuid,display,gender,age)'
         }).then(function(response) {
-            ctrl.relatedPersons = response.results;
+            ctrl.relatedPersons = [];
+            filterSearchedPersons(response.results);
         });
     }
 
+    function filterSearchedPersons(searchedPersons) {
+        if (angular.isArray(searchedPersons) && searchedPersons.length > 0) {
+            angular.forEach(searchedPersons, function(person) {
+                if( !relationshipsContain( person.uuid ) ) {
+                    // when searching for persons to create new relationships exclude the persons that already have a relationship with this person
+                    ctrl.relatedPersons.push(person);
+                }
+            });
+        }
+    }
+
     function getRelationshipTypes(types) {
         angular.forEach(types, function (type) {
-            if (findRelTypeByName(type.aIsToB) == null) {
-                var relTypeA = {};
-                relTypeA.uuid = type.uuid;
-                relTypeA.name = type.aIsToB;
-                relTypeA.type = "B";
-                ctrl.types.push(relTypeA);
-            }
-            if (findRelTypeByName(type.bIsToA) == null) {
-                var relTypeB = {};
-                relTypeB.uuid = type.uuid;
-                relTypeB.name = type.bIsToA;
-                relTypeB.type = "A";
-                ctrl.types.push(relTypeB);
+            if ((ctrl.allowedTypes.length < 1) || ((ctrl.allowedTypes.length > 0)
```

Comment:
Thanks @mogoodrich . I was trying to say that if a relationship type filter was not specified then we display all types, OR if a filter was defined then we only display the types included in the filter. I will add more comments and put the second OR statement on the same line.